Critical CVE policy (#593)
chainguard-dev/edu: content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/critical-cve-policy.md

---
title: "Ensure there are no High or Critical CVEs in your workloads
"
type: "article"

> description: "Understanding Identity and Access Management in Chainguard Enforce"

Reviewer: This description seems not right?

Author: Whoops! Fixed!

date: 2023-04-12T15:22:20+01:00
lastmod: 2023-04-12T15:22:20+01:00
draft: false
images: []
tags: ["policy-controller", "Policies", "Enforce", "Product"]
menu:
  docs:
    parent: "chainguard-enforce-kubernetes"
weight: 820
toc: true
---

> While Common Vulnerabilities and Exposures (CVEs) are undesirable at any time, the security standards of some industries strictly regulate the allowance of _high_ or _critical_ CVEs in industry-specific software.  For example, in the payment industry, the [PCI Security Standards Council](https://www.pcisecuritystandards.org/) requires that all vulnerabilities with a Common Vulnerability Scoring System (CVSS) score higher than 4 are addressed.

Reviewer: What do we mean by the vulnerabilities "are addressed"?

```suggestion
While Common Vulnerabilities and Exposures (CVEs) are undesirable at any time, the software security standards of certain industries strictly regulate the allowance of _high_ or _critical_ CVEs.  For example, in the payment industry, the [PCI Security Standards Council](https://www.pcisecuritystandards.org/) requires that all vulnerabilities with a Common Vulnerability Scoring System (CVSS) score higher than 4 are addressed.
```

Author: That was language Matt Moore used elsewhere. I think it refers to the fact that these CVEs are not entirely prohibited, they just have to be investigated to make sure they don't pose a real risk. 

Would switching "addressed" to "resolved" work? I'm not sure I want to get too off topic by explaining the above, but I also don't want to confuse readers!

Reviewer: "Addressed" is fine but it seems like it's missing something, like a time horizon or something

> For engineers and security professionals working in these contexts, it’s essential to know if images have high or critical CVEs before deploying them. But tracking these CVEs manually can be difficult, especially when regularly pulling or updating large numbers of images for your workloads.

Reviewer: I thought having a "container" in there may help with people finding this resource

```suggestion
For engineers and security professionals working in these contexts, it’s essential to know if container images have high or critical CVEs before deploying them. But tracking these CVEs manually can be difficult, especially when regularly pulling or updating large numbers of images for your workloads.
```

> ## Policy solution: Vulnerability attestation with no High or Critical CVEs

Reviewer: I would be consistent in title case or not here

> One way of addressing this problem is to use Chainguard’s policy that checks an image’s attestation to determine whether the image has any high or critical CVEs. Used with an admissions controller (such as [Chainguard Enforce](https://www.chainguard.dev/chainguard-enforce) or the open source [Sigstore policy-controller](https://edu.chainguard.dev/open-source/sigstore/policy-controller/how-to-install-policy-controller)), this policy enables you to restrict images or receive a warning whenever an image fails to meet the policy’s requirements.

Reviewer: ```suggestion
One way of addressing this concern is to use Chainguard’s policy that checks an image’s attestation to determine whether the image has any high or critical CVEs. Used with an admissions controller (such as [Chainguard Enforce](https://www.chainguard.dev/chainguard-enforce) or the open source [Sigstore policy-controller](/open-source/sigstore/policy-controller/how-to-install-policy-controller)), this policy enables you to restrict images or receive a warning whenever an image fails to meet the policy’s requirements.
```

> For this policy to work, the image under inspection must have an attached attestation containing output from its vulnerability scan. These vulnerability attestations are typically generated by the upstream maintainer who inserts the output of an image’s vulnerability scan into a vulnerability attestation and then signs the attestation to assure downstream users of its integrity. If the image doesn’t have a vulnerability attestation or you want to double check the attestation, you can also create a vulnerability attestation yourself using a scanner like Trivy. In either case, the vulnerability attestation is checked by this policy to determine whether the image contains any high or critical vulnerabilities.

Reviewer: I think it's ok to link out to Trivy

```suggestion
For this policy to work, the image under inspection must have an attached attestation containing output from its vulnerability scan. These vulnerability attestations are typically generated by the upstream maintainer who inserts the output of an image’s vulnerability scan into a vulnerability attestation and then signs the attestation to assure downstream users of its integrity. If the image doesn’t have a vulnerability attestation or you want to double check the attestation, you can also create a vulnerability attestation yourself using a scanner like [Trivy](https://trivy.dev/). In either case, the vulnerability attestation is checked by this policy to determine whether the image contains any high or critical vulnerabilities.
```

                warnMsg = ""
                response := {
                  "result" : result,
                  "error" : errorMsg,
                  "warning" : warnMsg
                }
              }
```

## Implementing this policy

You can use this policy freely with the open source [Sigstore policy-controller](https://edu.chainguard.dev/open-source/sigstore/policy-controller/how-to-install-policy-controller) to block new deployments of images that don’t meet the policy’s requirements.

However, if you are interested in making sure both incoming and _already admitted_ images meet this policy, you may be interested in [Chainguard Enforce](https://www.chainguard.dev/chainguard-enforce). With Chainguard Enforce, all of your images will be continuously evaluated against your selected policies, even if they passed these policies when initially admitted. You can set Enforce to send a notification (including posting to Slack, or opening a GitHub issue) if and when a deployed image falls out of compliance so that corrective action may be taken.

> To learn more about Chainguard Enforce, you can read the [Enforce Overview](https://edu.chainguard.dev/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/enforce-overview/) and browse the [Enforce documentation](https://edu.chainguard.dev/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/). You can also request access to the product by selecting Chainguard Enforce for Kubernetes on the [inquiry form](https://www.chainguard.dev/contact?utm_source=docs).

Reviewer: ```suggestion
To learn more about Chainguard Enforce, you can read the [Enforce Overview](/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/enforce-overview/) and browse the [Enforce documentation](/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/). You can also request access to the product by selecting Chainguard Enforce for Kubernetes on the [inquiry form](https://www.chainguard.dev/contact?utm_source=docs).
```
